### PR TITLE
Default support

### PR DESCRIPTION
**Pull request by Jim8y**
This pr add support to default syntax.

### CODE REVIEW

File: src/Neo.Compiler.CSharp/CompilationEngine.cs [hunk 1]
```diff
@@ -34,7 +35,7 @@ public class CompilationEngine
         private static readonly MetadataReference[] CommonReferences;
         private static readonly Dictionary<string, MetadataReference> MetaReferences = new();
         private static readonly Regex s_pattern = new(@"^(Neo\.SmartContract\.Framework\.SmartContract|SmartContract\.Framework\.SmartContract|Framework\.SmartContract|SmartContract)$");
-        internal readonly Dictionary<INamedTypeSymbol, CompilationContext> Contexts = new(SymbolEqualityComparer.Default);
+        internal readonly ConcurrentDictionary<INamedTypeSymbol, CompilationContext> Contexts = new(SymbolEqualityComparer.Default);
```
Comment (shargon) [inline src/Neo.Compiler.CSharp/CompilationEngine.cs:38]: This was already made, isn't it?

Comment (Jim8y) [reply src/Neo.Compiler.CSharp/CompilationEngine.cs:38]: previously it is dicionary then lock,,,, did i made this change somewhere else?

Comment (shargon) [reply src/Neo.Compiler.CSharp/CompilationEngine.cs:38]: Maybe I'm wrong, but I remember that you changed it in somewhere else

Comment (Jim8y) [reply src/Neo.Compiler.CSharp/CompilationEngine.cs:38]: > Maybe I'm wrong, but I remember that you changed it in somewhere else

the base is master, maybe changed in some unmerged pr,,,, my memory is bad.....if you think it is better, i can revert the change.

File: src/Neo.Compiler.CSharp/CompilationEngine.cs [hunk 2]
```diff
@@ -163,15 +163,18 @@ private List<CompilationContext> CompileProjectContracts(Compilation compilation
...
-            {
-                var dependencies = classDependencies.TryGetValue(c, out var dependency) ? dependency : new List<INamedTypeSymbol>();
-                var classesNotInDependencies = allClassSymbols.Except(dependencies).ToList();
-                var context = new CompilationContext(this, c, classesNotInDependencies!);
-                context.Compile();
-                // Process the target contract add this compilation context
-                Contexts.TryAdd(c, context);
-            });
+
+            Task.WhenAll(
```
[the remark this answers is not in the document]
Comment (Jim8y) [reply src/Neo.Compiler.CSharp/CompilationEngine.cs:167]: running in parallel is faster in out local machine, isn't it?  There are hundreds of contracts to be compiled, need to be faster.

[the remark this answers is not in the document]
Comment (shargon) [reply src/Neo.Compiler.CSharp/CompilationEngine.cs:167]: But you create a task per entry, is not managed, with parallel it's controlled according your cores

[the remark this answers is not in the document]
Comment (Jim8y) [reply src/Neo.Compiler.CSharp/CompilationEngine.cs:167]: it does not really matter actually, system will assign resources to tasks.

**Review by shargon — APPROVED** [review]

**Pull request merged**